src/charm.py

```diff
@@ -644,22 +646,20 @@ def generate_relation_databases(self) -> Dict[str, Dict[str, Union[str, bool]]]:
         """Generates a mapping between relation and database and sets it in the app databag."""
         if not self.unit.is_leader():
             return {}
-        if dbs := self.get_relation_databases():
-            return dbs
```

Comment:
Regenerating should be fast enough and if the mapping is wrong for whatever reason, always using the cache will never fix it.